amanuensis/resources/admin/admin_associated_user.py

```diff
@@ -29,6 +30,16 @@ def get_codes_for_roles():
         roles = udm.get_associated_user_roles(session)
         return {data.role: data.code for data in roles}
 
+def update_associated_user_user_id(logged_user_id, logged_user_email):
+    with flask.current_app.db.session as session:
+        user = udm.associate_user.get_associated_user(session, logged_user_email)
+        if not user:
+            raise NotFound("Associated User does not exist") 
+        if not user.user_id:
+            user.user_id = logged_user_id
+            return udm.associate_user.update_associated_user(session, user)
```

Comment:
the `update_associated_user` fetch again the value of the associated user from the DB/ORM. Without a flush before calling this line I am not sure the user_id that has just been updated the row above will be picked up, and if you do a flush you don't need to call the `udm.associate_user.update_associated_user` function. So probably just call the flush here. In general just check this is persisted.